app/grandchallenge/evaluation/forms.py

```diff
@@ -539,3 +547,24 @@ def clean(self):
             raise ValidationError("This challenge has exceeded its budget")
 
         return cleaned_data
+
+
+class ConfigureAlgorithmPhasesForm(SaveFormInitMixin, Form):
+    phases = ModelMultipleChoiceField(
```

Comment:
Shouldn't the phases be filtered by the current challenge?

Reply:
Ah wait, you have it on the challenges list, this might be better on the per-challenge level (e.g. url in the evaluation app and link from a page in there). I don't see a reason to do this for multiple challenges at once? Maybe I'm missing something. That way you can also add the time limit from the challenge request in the form.

Reply:
What current challenge? I implemented the view outside of the challenge context.

Reply:
Sorry, your comment somehow wasn't showing when I wrote mine. Ok, I think you're right that we mostly want to configure phases from one and the same challenge and never across challenges. So yes, moving this to the challenge level makes sense. I'm only wondering, from a usability perspective, staff users don't have access to the challenge pages, so a link on those pages won't be visible to us. We'll have to go straight to the url to do this then.